src/borg/remote.py

```diff
@@ -173,10 +174,15 @@ def __init__(self, location, create=False, lock_wait=None, lock=True, args=None)
             version = self.call('negotiate', RPC_PROTOCOL_VERSION)
         except ConnectionClosed:
             raise ConnectionClosedWithHint('Is borg working on the server?') from None
-        if version != RPC_PROTOCOL_VERSION:
+        if version < RPC_PROTOCOL_DOWNGRADABLE_VERSION or version > RPC_PROTOCOL_VERSION:
             raise Exception('Server insisted on using unsupported protocol version %d' % version)
         try:
-            self.id = self.call('open', self.location.path, create, lock_wait, lock)
+            if version >= 3:
+                self.id = self.call('open', self.location.path, create, lock_wait, lock, append_only)
+            else:
+                if append_only:
+                    raise Exception('Server (protocol version %d) does not support --append-only' % version)
```

Comment:
Raising an Exception means that it generates a very ugly error message. Do we have an alternative? The same thing already exists above for an incompatible version.